RUFAS/report_generator.py

```diff
@@ -133,11 +136,33 @@ def subtraction_aggregator(data: List[float]) -> float:
 
 
 class ReportGenerator:
+    """
+    A class to generate reports based on filtered data and aggregation criteria and store them in a dictionary.
+    """
+
+    def __init__(self):
+        """
+        Initializes the ReportGenerator.
+        """
+
+        self.reports: Dict[str, Dict[str, List[Any]]] = {}
+
+    def clear_reports(self):
+        """
+        Clears the reports generated by the ReportGenerator.
+
+        Returns
+        -------
+        None
+        """
+
+        self.reports = {}
+
     def generate_report(
-        self,
-        filtered_pool: Dict[str, Dict[str, List[Any]]],
-        filter_content: Dict[str, str | int | List[str]],
-    ) -> Dict[str, List[float]]:
+            self,
+            filtered_pool: Dict[str, Dict[str, List[Any]]],
+            filter_content: Dict[str, str | int | List[str]],
+    ) -> List[float]:
```

Comment:
```suggestion
    ) -> Dict[str, List[float]]:
```
This should be changed back or the values being returned should be changed.

Reply:
Right, the type here should be `Dict[str, List[float]]`.